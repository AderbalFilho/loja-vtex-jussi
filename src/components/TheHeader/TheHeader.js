import { JIcon, JInput, JLink } from '@ui';
import TheNavbar from '@components/TheNavbar';

import jussiLogo from '@assets/img/logoJussiVectorGreen.svg';

import './TheHeader.scss';

function TheHeader() {
  const links = [
    { url: 'https://jussi.com.br/services.html', label: 'Nossas soluções' },
    { url: 'https://jussi.com.br/', label: 'Conheça a Jüssi' },
  ];

  return (
    <header className="the-header">
      <div className="the-header__left">
        <JLink href="#">
          <img src={jussiLogo} alt="Logo Jüssi - The business agency" />
        </JLink>
        <TheNavbar links={links} />
      </div>
      <div className="the-header__right">
        <JInput
          classesContainer="the-header__input"
          type="text"
          placeholder="Buscar"
          icon="search"
        />
        <JLink classes="the-header__login" href="#">
          Login
        </JLink>
        <JIcon icon="shoppingCart" />
      </div>
    </header>
  );
}

export default TheHeader;
