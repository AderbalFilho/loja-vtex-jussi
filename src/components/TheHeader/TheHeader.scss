.the-header {
  display: flex;
  margin: 20px 24px;
  justify-content: space-between;
}

.the-header__left {
  align-items: center;
  display: flex;
}

.the-header__right {
  align-items: center;
  display: flex;
}

.the-header__input {
  margin-right: 32px;
}

.the-header__login {
  margin-right: 32px;
}
