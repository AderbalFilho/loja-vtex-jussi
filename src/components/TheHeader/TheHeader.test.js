import React from 'react';
import TestRenderer from 'react-test-renderer';

import TheHeader from './TheHeader';

describe('The Header', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<TheHeader />);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
