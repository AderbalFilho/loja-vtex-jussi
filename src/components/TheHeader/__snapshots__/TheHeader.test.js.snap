// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header matches snapshot 1`] = `
<header
  className="the-header"
>
  <div
    className="the-header__left"
  >
    <a
      className="j-link"
      href="#"
    >
      <img
        alt="Logo Jüssi - The business agency"
        src=""
      />
    </a>
    <nav
      className="the-navbar"
    >
      <a
        className="j-link the-navbar__link"
        href="https://jussi.com.br/services.html"
        rel="noreferrer noopener"
        target="_blank"
      >
        Nossas soluções
      </a>
      <a
        className="j-link the-navbar__link"
        href="https://jussi.com.br/"
        rel="noreferrer noopener"
        target="_blank"
      >
        Conheça a Jüssi
      </a>
    </nav>
  </div>
  <div
    className="the-header__right"
  >
    <div
      className="j-input the-header__input"
    >
      <input
        className="j-input__input j-input__input--with-icon"
        onChange={[Function]}
        placeholder="Buscar"
        type="text"
        value=""
      />
      <img
        alt="buscar"
        className="j-input__icon"
        onClick={[Function]}
        src=""
      />
    </div>
    <a
      className="j-link the-header__login"
      href="#"
    >
      Login
    </a>
    <img
      alt="carrinho de compras"
      className="j-icon"
      src=""
    />
  </div>
</header>
`;
